feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -217,7 +221,13 @@ fun PreviewScreen(
                     val multipleEventsCutter = remember { MultipleEventsCutter() }
                     /*todo: close quick settings on start record/image capture*/
                     CaptureButton(
+                        modifier = Modifier
+                            .testTag(CAPTURE_BUTTON),
                         onClick = {
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                                // this trace is closed at the imageCaptureUseCase callback in CameraxCameraUseCase
+                                Trace.beginAsyncSection(IMAGE_CAPTURE_TRACE, 0)
```

Comment:
It's a good exercise to see how `beginAsyncSection` / `endAsyncSection` actually work, but we can make this more self-contained with kotlin/coroutines.

Rather than starting this in the `feature` module and ending it in the `domain` module, we can do this all in a single place. If you move this into `PreviewViewModel.captureImage()`, you can change the `viewModelScope.launch` block to contain:
```
traceAsync(IMAGE_CAPTURE_TRACE, 0) {
  try {
    cameraUseCase.takePicture()
  } catch ... {
    ...
  }
}
```

The [`traceAsync`](https://developer.android.com/reference/kotlin/androidx/tracing/package-summary#traceAsync(kotlin.String,kotlin.Int,kotlin.coroutines.SuspendFunction0)) function comes from `androidx.tracing:tracing-ktx`.

This does slightly change the section that is being traced (it starts inside `viewModelScope` rather than in the `onClick` callback), so there are some other improvements we could make on top of it, but it would be a good start to make the code cleaner.

This will also allow us in the future to update the cookie to be unique so we can trace multiple captures at the same time.